File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/documentdialog/DocumentDialogMvpView.java
package com.mifos.mifosxdroid.dialogfragments.documentdialog;

import com.mifos.api.GenericResponse;
import com.mifos.mifosxdroid.base.MvpView;

/**
 * Created by Rajan Maurya on 8/6/16.
 */
public interface DocumentDialogMvpView extends MvpView {

    void checkPermissionAndRequest();

    void requestPermission();

    void getExternalStorageDocument();

    void showDocumentedCreatedSuccessfully(GenericResponse genericResponse);

    void showDocumentUpdatedSuccessfully();

    void showError(int errorMessage);

    void showUploadError(String errorMessage);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/documentdialog/DocumentDialogMvpView.kt
package com.mifos.mifosxdroid.dialogfragments.documentdialog

import com.mifos.api.GenericResponse
import com.mifos.mifosxdroid.base.MvpView

/**
 * Created by Rajan Maurya on 8/6/16.
 */
interface DocumentDialogMvpView : MvpView {
    fun checkPermissionAndRequest()
    fun requestPermission()
    fun getExternalStorageDocument()
    fun showDocumentedCreatedSuccessfully(genericResponse: GenericResponse)
    fun showDocumentUpdatedSuccessfully()
    fun showError(errorMessage: Int)
    fun showUploadError(errorMessage: String)
}